### ISSUE TITLE
sub-generators create inconsistent code

### ISSUE THREAD

**Issue by jluna79** [thardy/generator-ngbp#21]
When creating a submodule such as: 

`yo ngbp:module "browse.profile"`

the files are correctly created under `app/browse/profile`. However there are a few inconsistencies.

In `app.js`:
* The generator names the module `browseProfile` but it is referenced as: `appName.browse.profile`.

In `profile.module.js`:
* The `url` value is `/browse-profile` but it should be `/profile`.
* The templateUrl is set to `browse/profile/browse-profile.tpl.html` however the generated file is called `profile.tpl.html`.
* Not an inconsistency but the value for `$stateProvider.state` is set to `browse-profile`. It would be more ad hoc to the ui-router if it was `browse.profile` (dot-separated).

I also created the same submodule (browse.profile) using the RESTful option and got the following:

In `profile.js`:
* The value in the model is created as `model.browse/profile = [];`. The slash ("/") produces an error when excecuting jshint.

I'm using generator-ngbp version 2.9.1

Hope this helps

**Comment by thardy** [comment]
I'll have to take a look at these.  The only thing I can comment on off the top of my head is that we cannot use dot-separated state names with ui-router.  The dots mean inheritance to ui-router, and they will cause problems if you aren't meaning to embed the new route as a child state of another route.

As to the others, if you could give examples of exactly what you think the strings should be in each of these cases, I'll go through and fix them.

**Comment by jluna79** [comment]
Sure thing, here it goes:

**In `app.js`:**
> * The generator names the module `browseProfile` but it is referenced as: `appName.browse.profile`.

The generated code should be:
```
...
}(angular.module("appName", [
    'tantanApp.browseProfile',
])));
```

**In `profile.module.js`:**
> * The `url` value is `/browse-profile` but it should be `/profile`.

This only applies when the submodule is, in fact, part of the state-machine (the dot-separated topic you answered in your previous comment) so it is not longer an inconsistency.

> * The templateUrl is set to `browse/profile/browse-profile.tpl.html` however the generated file is called `profile.tpl.html`.

Here the correct code should be:

```
            views: {
                "main": {
                    controller: 'BrowseProfileController as model',
                    templateUrl: 'browse/profile/profile.tpl.html'
                }
            },
```

> * Not an inconsistency but the value for `$stateProvider.state` is set to `browse-profile`. It would be more ad hoc to the ui-router if it was `browse.profile` (dot-separated).

Answered by you above and I completely agreed. After doing some more tests I found that it is more common having submodules that have nothing to do with the current state of the ui-router.

> I also created the same submodule (browse.profile) using the RESTful option and got the following:

> In `profile.js`:
> * The value in the model is created as `model.browse/profile = [];`. The slash ("/") produces an error when excecuting jshint.

In order to maintain naming consistent the model variable should be `model.browseProfile = [];`

Cheers

**Comment by thardy** [comment]
I took a stab at this.  I didn't name everything exactly as you suggested, but it's close.  The differences are with the url - for "browse.profiles" (I like to use plural module names) I went with "browse/profiles" as the url because a submodule seems like it should be contextual most of the time.  It's easily changed if you make one that shouldn't be.

While I kept the "<appName>.browseProfiles" as the module name, I went with an internal usage of just  "profiles" (e.g. "browse/profiles/addProfile.tpl.html" for add url, and "model.profiles" as the var in profiles.js.

Let me know what you think.  I figure there's no way I can satisfy everyone, but hopefully that's close  enough to what most people want that minor revisions won't be a big deal.
